src/lib.rs

```diff
@@ -159,6 +161,25 @@ pub enum SanType {
 	IpAddress(IpAddr),
 }
 
+#[cfg(feature = "x509-parser")]
+fn ip_addr_from_octets(octets: &[u8]) -> Result<IpAddr, RcgenError> {
+	match <&[u8; 16]>::try_from(octets) {
```

Comment:
Can you put back the old matching on the length of octets? That needs an unwrap but it seems nicer to me than this code.

Reply:
I can do that but some people take issue with `unwrap` in library (non-test, non-example) code.

Another way besides `match` you may like better would be `if let`, for example

```
	if let Ok(ipv6_octets) = <&[u8; 16]>::try_from(octets) {
		Ok(Ipv6Addr::from(*ipv6_octets).into())
	}
	else if let Ok(ipv4_octets) = <&[u8; 4]>::try_from(octets) {
		Ok(Ipv4Addr::from(*ipv4_octets).into())
	}
	else {
		Err(RcgenError::InvalidIpAddressOctetLength(octets.len()))
	}
```

Thoughts on the `if let` variant?

Reply:
I don't mind unwraps in this location, but if you prefer if let that's also fine. Both code looks nice, but a nested match just isn't beautiful.

Reply:
`if let` pushed